test/cxx/Core/ControllerTest.cpp

```diff
@@ -1144,4 +1144,66 @@ namespace tut {
 		string header = readResponseHeader();
 		ensure(containsSubstring(header, "HTTP/1.1 502"));
 	}
+
+	/***** Application async connection *****/
+
+	TEST_METHOD(60) {
+		set_test_name("Testing async connect flow.");
+		init();
+		mockNextSession();
+		struct ev_async& async = testSession.setupAsync(bg.libev_loop);
+
+		connectToServer();
+		sendRequest(
+			"GET /hello HTTP/1.1\r\n"
+			"Host: localhost\r\n"
+			"Connection: close\r\n"
+			"\r\n");
+	    EVENTUALLY(5, { result = async.data != NULL; });
+		ev_async_send(bg.libev_loop, &async);
+		waitUntilSessionInitiated();
+    }
+
+	TEST_METHOD(61) {
```

Comment:
This test strategy does not work. You are delaying the accept() until ev_async_send(). But the connect() still finishes immediately (i.e. the ev_io watcher on the socket still gets an EV_WRITE event soon, before the timeout fires).